### PR TITLE
feat: add release information args to proguard mapping upload task

### PR DESCRIPTION
**Pull request by buenaflor**
## :scroll: Description
Release information can now be attached to the `sentry-cli` call during the proguard mapping upload task. 

`sentry-cli` already has a mechanism to add those args: `--app-id, --version, --version-code` so no work there needs to be done right now.

## :bulb: Motivation and Context

This is needed for associating a release to proguard mapping files. See https://github.com/getsentry/sentry-android-gradle-plugin/issues/40

## :green_heart: How did you test it?
manually with different flavors and unit test

## :pencil: Checklist

- [ ] I reviewed the submitted code
- [ ] I added tests to verify the changes
- [ ] I updated the docs if needed
- [ ] No breaking changes

### CODE REVIEW

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/util/ReleaseInfo.kt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package io.sentry.android.gradle.util
+
+data class ReleaseInfo(
+    val applicationId: String,
+    val versionName: String,
+    val versionCode: Int? = null
+)
```
Comment (romtsn) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/util/ReleaseInfo.kt:7]:
```suggestion
) : Serializable
```
`m`: I think it has to be serializable if we pass it as a task input

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/AndroidComponentsConfig.kt [hunk 2]
```diff
@@ -238,3 +241,18 @@ private fun AndroidComponentsExtension<*, *, *>.configureVariants(callback: (Var
         onVariants70(this, callback)
     }
 }
+
+private fun getReleaseInfo(project: Project, variant: Variant): ReleaseInfo {
+    val appExtension = project.extensions.getByType(AppExtension::class.java)
+    var applicationId =
+        appExtension.defaultConfig.applicationId ?: appExtension.namespace.toString()
+    var versionName = appExtension.defaultConfig.versionName ?: "1.0.0"
```
Comment (romtsn) [inline plugin-build/src/main/kotlin/io/sentry/android/gradle/AndroidComponentsConfig.kt:249]:
```suggestion
    var versionName = appExtension.defaultConfig.versionName ?: "undefined"
```
Maybe that is a bit more explicit than `1.0.0`? Idk if we have server-side validation of semver though, but I'd try to send undefined and see what happens

**Review by romtsn — APPROVED** [review]
lgtm besides some minor comments 👍

File: plugin-build/src/main/kotlin/io/sentry/android/gradle/AndroidComponentsConfig.kt (hunk 2 above)
Comment (buenaflor) [reply plugin-build/src/main/kotlin/io/sentry/android/gradle/AndroidComponentsConfig.kt:249]: After implementing the backend, we don't have a semver validation here so we're I guess we're free to pass in what makes sense in that case. I assume `undefined` seems reasonable if a version can't be determined.

**Comment by buenaflor** [comment]
@romtsn @markushi do you know what I changed that might make the integration test fail? or is it flakiness? 

It seems like all tests are failing of versions [AGP 7.3.0, Gradle 7.5] or lower

**Pull request merged**